pkg/compute/executor_buffer.go

```diff
@@ -141,13 +142,13 @@ func (s *ExecutorBuffer) doRun(ctx context.Context, task *bufferTask) {
 
 	select {
 	case <-ctx.Done():
-		s.callback.OnComputeFailure(ctx, ComputeError{
+		log.Ctx(ctx).Info().Str("ID", task.execution.ID).Dur("Timeout", timeout).Msg("Execution timed out")
+		s.callback.OnCancelComplete(ctx, CancelResult{
```

Comment:
I'm not entirely convinced that a cancellation of context necessarily means the job was canceled, so I'm hesitant to suggest we modify this logic right away. E.g. there could be a situation where the compute node is shut down. In this case, the context will be canceled, not because of a timeout, but due to a failure in the compute node's operations. It's like if I, acting as a compute provider, decide to shut down my node gracefully, causing your job to stop running. You'd naturally assume it failed, yea?

Also, if a user expects their job to be finished in 5 minutes and it doesn't, I'd be inclined to label that job as 'Failed.' To me, a canceled job is something initiated directly by the user through the API (Our [code comments](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/model/execution_state.go#L32) kind of suggest this). It's not something that happens indirectly through a timeout specified in the job spec. Wdyt?

Reply:
There's no modification of the logic here – the existing code already assumes that a context cancellation is a timeout (see the error message). The context is only ever the background atm: 
https://github.com/bacalhau-project/bacalhau/blob/07d2c37a40630c70d286d99511c2ceaee1eabcd4/pkg/compute/executor_buffer.go#L187

This would be OK if the delegate executor was guaranteed to stop the job on node shutdown. But I can't see any logic that does this. So this does look like a bug – I can't see any logic anywhere for ending jobs when the node shuts down? However, I think that's a bigger bug than I am trying to handle here. So I'd prefer to leave this as is for now and investigate that issue separately.